joss-paper/paper.md

```diff
@@ -0,0 +1,123 @@
+identify regions of ionized hydrogen atop a cosmological density field evolved using
+first- or second-order perturbation theory.
+However, `21cmFAST` is a highly specialized code, and as such has suffered from its
+difficulty of use.
+This is more serious than the simple discomfort of end-users (or the reduction in
+end-user-ship that it entails); the lack of modularity within the code has led to
+widespread code "branching" as researchers hack new physical features of interest
+into the C code; the lack of a streamlined API has led derivative codes which run
+multiple realizations of `21cmFAST` simulations (such as the Monte Carlo simulator,
+`21CMMC`) to re-write large portions of the code in order to serve their purpose.
+It is thus of critical importance, as the field moves forward in its understanding -- and
+the range and scale of physical models of interest continues to increase -- to
+reformulate the `21cmFAST` code in order to provide a fast, modular, well-documented,
+well-tested, stable simulator for the community.
+
+
+In `21cmFAST` v3+, which this paper presents, this goal has come a significant way towards
+achievement.
+While keeping the same core functionality of previous versions of `21cmFAST`, it has
+been fully integrated into a Python package, with a simple and intuitive interface, and
+a great deal more flexibility.
+At a higher level, in order to maintain best practices, a community of users and
+developers has coalesced into a formal collaboration which maintains the project via a
+Github organization.
+This allows the code to be consistently monitored for quality, maintaining high test
+coverage, stylistic integrity, dependable release strategies and versioning,
+and peer code review.
+It also provides a single point-of-reference for the community to obtain the code,
+report bugs and request new features (or get involved in development).
+
+A significant part of the work of moving to a Python interface has been the
+development of a robust series of underlying Python structures which handle the passing
+of data between Python and C via the `CFFI` library.
+This foundational work provides a platform for future versions to extend the scientific
+capabilities of the underlying simulation code.
+The primary *new* usability features of `21cmFAST` v3+ are:
+
+* Convenient (Python) data objects which simplify access to and processing of the various
+  fields that form the brightness temperature.
+* Enhancement of modularity: the underlying C functions for each step of the simulation
+  have been de-coupled, so that arbitrary functionality can be injected into the process.
+* Simple `pip`-based installation.
+* Robust on-disk caching/writing of data, both for efficiency and simplified reading of
+  previously processed data (using HDF5).
+* Simple high-level API to generate either coeval cubes or full lightcone data.
+* Improved exception handling and debugging.
+* OpenMP parallel processing.
```

Comment:
OpenMP parallelization was in 21cmfast for years...

Reply:
Yeah, I guess it's something we took out then put back in.